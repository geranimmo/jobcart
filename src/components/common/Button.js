import React from "react";

const Button = props => {
	return <button type="button" {...props}>{props.value}</button>;
};

export { Button };
