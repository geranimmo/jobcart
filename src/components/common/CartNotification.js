import React from "react";

const CartNotification = props => {
	const {
		totalItems,
		className
	} = props;
	const cartTotal = totalItems > 9 ? '9+' : totalItems;
    
	return <div className={`${className} ${totalItems === 0 ? 'hidden' : ''}`}>{cartTotal}</div>;
};

export { CartNotification };
