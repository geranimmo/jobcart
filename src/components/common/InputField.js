import React from "react";

const InputField = props => {
	return <input {...props}/>;
};

export { InputField };
