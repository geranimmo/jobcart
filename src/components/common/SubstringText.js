const SubstringText = props => {
	return props.text.length > props.textMaxLength ?
		props.text.substring(0, props.textMaxLength)+'...'
		:
		props.text;
};

export { SubstringText };