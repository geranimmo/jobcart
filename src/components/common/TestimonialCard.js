import React from 'react';

const TestimonialCard = props => {
	const { TestimonialWrapper } = styles;
    
	return (
		<div style={TestimonialWrapper} className={props.className}>
			{props.children}
		</div>
	);
};

// sample of inline style
const styles = {
	TestimonialWrapper: {
		display: 'inline-block'
	}
};

export { TestimonialCard };