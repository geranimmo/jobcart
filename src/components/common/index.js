export { InputField } from './InputField';
export { Button } from './Button';
export { CartFooter } from './CartFooter';
export { DateTimeConverter } from './DateTimeConverter';
export { SubstringText } from './SubstringText';
export { CartItem } from './CartItem';
export { CartNotification } from './CartNotification';