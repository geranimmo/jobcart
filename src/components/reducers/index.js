import { combineReducers } from 'redux';
import LoginReducer from './LoginReducers';
import ProfileReducer from './ProfileReducers';
import ShoppingCartReducer from './ShoppingCartReducers';
import PackageReducer from './PackageReducers';
import TestimonialReducer from './TestimonialReducers';
import TotalCostReducer from './TotalCostReducers';
import TotalDiscountReducer from './TotalDiscountReducers';

export default combineReducers({
	UserLogin: LoginReducer,
	Profile: ProfileReducer,
	ShoppingCart: ShoppingCartReducer,
	Packages: PackageReducer,
	Testimonials: TestimonialReducer,
	TotalCost: TotalCostReducer,
	TotalDiscount: TotalDiscountReducer
});
