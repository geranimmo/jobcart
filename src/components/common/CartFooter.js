import React from 'react';
import { Button } from './Button';

const CartFooter = props => {
	let fixedCost = parseFloat((props.totalCartCost - props.totalCartDiscount).toFixed(2));
	
	const buttonProps = {
		className: 'button__checkout',
		value: `Checkout (${props.myCartList.length})`
	};
	
	return (
		<div className={props.className}>
			<div>
				<span>
					<label>Total Cost:</label>
					${fixedCost}
				</span>
				<Button {...buttonProps}/>
			</div>
		</div>
	);
};

export { CartFooter };