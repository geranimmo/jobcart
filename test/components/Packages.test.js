import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import PackageSlider from '../../src/components/packages';
import Slider from 'react-slick';
import { Button } from '../../src/components/common/Button';
import reducers from '../../src/components/reducers';
import ListPackages from '../../src/assets/json/packages__.json';

describe('>>> P A C K A G E S  L I S T ---- Test & Snapshot <<<', () => {
	const store = createStore(reducers, {}, applyMiddleware(thunk));
	let mountWrapper, addToCart;

	beforeEach(() => {
		addToCart = jest.fn();
		mountWrapper = mount(
			<Provider store={store}>
				<Router>
					<PackageSlider
						addToCart={addToCart}
						packagesList={ListPackages}
					/>
				</Router>
			</Provider>
		);
	});

	it('+++ Render all Sliders component +++', () => {
		const finder = mountWrapper.find('.slider__container');
		expect(finder.length).toEqual(1);

		const sliderNavWrapper = finder.find('.slider__nav__wrapper');
		expect(sliderNavWrapper.length).toEqual(1);
		const sliderNavContent = sliderNavWrapper.find(Slider);
		expect(sliderNavContent.length).toEqual(1);

		const sliderContentWrapper = finder.find('.slider__content__wrapper');
		expect(sliderContentWrapper.length).toEqual(1);
		const sliderContent = sliderContentWrapper.find(Slider);
		expect(sliderContent.length).toEqual(1);

		// total sliders should be 2
		const sliderContents = mountWrapper.find(Slider);
		expect(sliderContents.length).toEqual(2);
	});

	it('+++ Should show Button addToCart +++', () => {
		const finder = mountWrapper.find('.slider__container');
		const sliderContentWrapper = finder.find('.slider__content__wrapper');
		const sliderContent = sliderContentWrapper.find(Slider);
		const buttonWrapper = sliderContent.find('.slick-current');
		const buttonAddToCart = buttonWrapper.find(Button);
		expect(buttonAddToCart.length).toEqual(1);
	});

	// it('+++ Should initialize addToCart when Button clicked in active slide +++', () => {
	// 	const finder = mountWrapper.find('.slider__container');
	// 	const sliderContentWrapper = finder.find('.slider__content__wrapper');
	// 	const sliderContent = sliderContentWrapper.find(Slider);
	// 	const buttonWrapper = sliderContent.find('.slick-current');
	// 	const buttonAddToCart = buttonWrapper.find(Button);
		
	// 	buttonAddToCart.simulate('click');
	// 	expect(addToCart).toBe(
	// 		expect.anything()
	// 	);
	// });

	it('+++ Capturing Snapshot of PackageSlider component +++', () => {
		const renderedValue = renderer.create(
			<Provider store={store}>
				<Router>
					<PackageSlider packagesList={ListPackages}/>
				</Router>
			</Provider>
		).toJSON();
		
		expect(renderedValue).toMatchSnapshot();
	});
});