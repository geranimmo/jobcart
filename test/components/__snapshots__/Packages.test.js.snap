// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> P A C K A G E S  L I S T ---- Test & Snapshot <<< +++ Capturing Snapshot of PackageSlider component +++ 1`] = `
<div
  className="slider__container "
>
  <div
    className="slider__nav__wrapper"
  >
    <div
      className="slick-slider slick-initialized"
    >
      
      <div
        className="slick-list"
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "WebkitTransform": "translate3d(0px, 0px, 0px)",
              "WebkitTransition": "",
              "msTransform": "translateX(0px)",
              "opacity": 1,
              "transform": "translate3d(0px, 0px, 0px)",
              "transition": "",
            }
          }
        >
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={0}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="slider__card__nav "
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <h1
                  className="no__margin"
                >
                  Classic
                </h1>
              </div>
            </div>
          </div>
          <div
            aria-hidden={false}
            className="slick-slide slick-active slick-current"
            data-index={1}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="slider__card__nav nav__recommended"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <h1
                  className="no__margin"
                >
                  Standout
                </h1>
              </div>
            </div>
          </div>
          <div
            aria-hidden={false}
            className="slick-slide slick-active"
            data-index={2}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="slider__card__nav "
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <h1
                  className="no__margin"
                >
                  Premium
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
  <div
    className="slider__content__wrapper"
  >
    <div
      className="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        className="slick-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={null}
        onMouseMove={null}
        onMouseUp={[Function]}
        onTouchCancel={null}
        onTouchEnd={[Function]}
        onTouchMove={null}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "WebkitTransform": "translate3d(0px, 0px, 0px)",
              "WebkitTransition": "",
              "msTransform": "translateX(0px)",
              "opacity": 1,
              "transform": "translate3d(0px, 0px, 0px)",
              "transition": "",
            }
          }
        >
          <div
            aria-hidden={true}
            className="slick-slide slick-cloned"
            data-index={-1}
            onClick={[Function]}
            style={
              Object {
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="section__carousel"
                itemProp="itemListElement"
                itemScope={true}
                itemType="http://schema.org/Product"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <div
                  className="slider__card__body"
                >
                  <img
                    alt="Premium"
                    className="slider__images"
                    src="/assets/images/employeer-image-sample3.jpg"
                  />
                  <h2
                    className="no__margin__top block width__100"
                  >
                    Premium Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry
                  </h2>
                  <span
                    className="label__info hidden"
                  >
                    RECOMMENDED
                  </span>
                  <span
                    className="pricing block width__100"
                  >
                    $
                    394.99
                  </span>
                  <button
                    className="button__buy__items"
                    onClick={[Function]}
                    type="button"
                    value="Buy Ad"
                  >
                    Buy Ad
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={0}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="section__carousel"
                itemProp="itemListElement"
                itemScope={true}
                itemType="http://schema.org/Product"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <div
                  className="slider__card__body"
                >
                  <img
                    alt="Classic"
                    className="slider__images"
                    src="/assets/images/employeer-image-sample.jpg"
                  />
                  <h2
                    className="no__margin__top block width__100"
                  >
                    Classic Ad Lorem Ipsum is simply dummy text of the printing and typesetting
                  </h2>
                  <span
                    className="label__info hidden"
                  >
                    RECOMMENDED
                  </span>
                  <span
                    className="pricing block width__100"
                  >
                    $
                    269.99
                  </span>
                  <button
                    className="button__buy__items"
                    onClick={[Function]}
                    type="button"
                    value="Buy Ad"
                  >
                    Buy Ad
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden={false}
            className="slick-slide slick-active slick-current"
            data-index={1}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="section__carousel"
                itemProp="itemListElement"
                itemScope={true}
                itemType="http://schema.org/Product"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <div
                  className="slider__card__body"
                >
                  <img
                    alt="Standout"
                    className="slider__images"
                    src="/assets/images/employeer-image-sample2.jpg"
                  />
                  <h2
                    className="no__margin__top block width__100"
                  >
                    Standout Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry of Lorem Ipsum
                  </h2>
                  <span
                    className="label__info "
                  >
                    RECOMMENDED
                  </span>
                  <span
                    className="pricing block width__100"
                  >
                    $
                    322.99
                  </span>
                  <button
                    className="button__buy__items"
                    onClick={[Function]}
                    type="button"
                    value="Buy Ad"
                  >
                    Buy Ad
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={2}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="section__carousel"
                itemProp="itemListElement"
                itemScope={true}
                itemType="http://schema.org/Product"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <div
                  className="slider__card__body"
                >
                  <img
                    alt="Premium"
                    className="slider__images"
                    src="/assets/images/employeer-image-sample3.jpg"
                  />
                  <h2
                    className="no__margin__top block width__100"
                  >
                    Premium Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry
                  </h2>
                  <span
                    className="label__info hidden"
                  >
                    RECOMMENDED
                  </span>
                  <span
                    className="pricing block width__100"
                  >
                    $
                    394.99
                  </span>
                  <button
                    className="button__buy__items"
                    onClick={[Function]}
                    type="button"
                    value="Buy Ad"
                  >
                    Buy Ad
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide slick-cloned"
            data-index={3}
            onClick={[Function]}
            style={
              Object {
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="section__carousel"
                itemProp="itemListElement"
                itemScope={true}
                itemType="http://schema.org/Product"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <div
                  className="slider__card__body"
                >
                  <img
                    alt="Classic"
                    className="slider__images"
                    src="/assets/images/employeer-image-sample.jpg"
                  />
                  <h2
                    className="no__margin__top block width__100"
                  >
                    Classic Ad Lorem Ipsum is simply dummy text of the printing and typesetting
                  </h2>
                  <span
                    className="label__info hidden"
                  >
                    RECOMMENDED
                  </span>
                  <span
                    className="pricing block width__100"
                  >
                    $
                    269.99
                  </span>
                  <button
                    className="button__buy__items"
                    onClick={[Function]}
                    type="button"
                    value="Buy Ad"
                  >
                    Buy Ad
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide slick-cloned"
            data-index={4}
            onClick={[Function]}
            style={
              Object {
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="section__carousel"
                itemProp="itemListElement"
                itemScope={true}
                itemType="http://schema.org/Product"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <div
                  className="slider__card__body"
                >
                  <img
                    alt="Standout"
                    className="slider__images"
                    src="/assets/images/employeer-image-sample2.jpg"
                  />
                  <h2
                    className="no__margin__top block width__100"
                  >
                    Standout Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry of Lorem Ipsum
                  </h2>
                  <span
                    className="label__info "
                  >
                    RECOMMENDED
                  </span>
                  <span
                    className="pricing block width__100"
                  >
                    $
                    322.99
                  </span>
                  <button
                    className="button__buy__items"
                    onClick={[Function]}
                    type="button"
                    value="Buy Ad"
                  >
                    Buy Ad
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide slick-cloned"
            data-index={5}
            onClick={[Function]}
            style={
              Object {
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="section__carousel"
                itemProp="itemListElement"
                itemScope={true}
                itemType="http://schema.org/Product"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <div
                  className="slider__card__body"
                >
                  <img
                    alt="Premium"
                    className="slider__images"
                    src="/assets/images/employeer-image-sample3.jpg"
                  />
                  <h2
                    className="no__margin__top block width__100"
                  >
                    Premium Ad Lorem Ipsum is simply dummy text of the printing and typesetting industry
                  </h2>
                  <span
                    className="label__info hidden"
                  >
                    RECOMMENDED
                  </span>
                  <span
                    className="pricing block width__100"
                  >
                    $
                    394.99
                  </span>
                  <button
                    className="button__buy__items"
                    onClick={[Function]}
                    type="button"
                    value="Buy Ad"
                  >
                    Buy Ad
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
